www/pfSense-pkg-squid/files/usr/local/pkg/squid.inc

```diff
@@ -1092,11 +1092,19 @@ function squid_validate_auth($post, &$input_errors) {
 		switch ($auth_method) {
 			case 'ldap':
 				$user = trim($post['ldap_user']);
+				$ldaptype = trim($post['ldap_urltype']);
+				$ldapversion = trim($post['ldap_version']);
 				if (empty($user)) {
 					$input_errors[] = "'LDAP server user DN' is required.";
 				} elseif (!$user) {
 					$input_errors[] = "'LDAP server user DN' must be a valid DN.";
 				}
+				if ($ldaptype == 'starttls' && $ldapversion == '2') {
+					$input_errors[] = "'STARTTLS supported only by LDAP version 3.";
+				}
+				if (pfs_version_compare(false, 2.4, $g['product_version']) && $settings['ldap_urltype'] != 'standard') {
+					$input_errors[] = "'SSL/STARTTLS feature supported only by pfSense >= 2.5.";
```

Comment:
Not sure we should completely exclude this. It only doesn't work if they need to trust a self-signed CA. If their LDAP server is using a globally trusted certificate, like one from Let's Encrypt, then it would work even on 2.4.x.
It's probably enough to just warn about that in the option text.

Reply:
done